### ISSUE TITLE
write function to return all of the building information (along with documentation)

### ISSUE THREAD

**Issue by nicholasjhorton** [STAT325-S24/HolyokeUrbanRenewal#3]
This function (perhaps named `get_buildings()` should return a tibble with all of the filenames from files that end in `png` in https://github.com/STAT325-S24/HolyokeUrbanRenewal/tree/main/inst/extdata.

It should provide multiple columns to allow display and merging of results with the spreadsheet/dataset saved as `HolyokeUrbanRenewal` (e.g., `block`, `parcel`, `type`, `file_number`).

This can likely be done in parallel with #2

**Comment by nicholasjhorton** [comment]
@arogers24 and @tknightly24 can you please update us on your work to date? I thought that you had stubbed out a function that we could review in terms of API. Thanks in advance for any guidance, Nick

**Comment by nicholasjhorton** [comment]
@arogers24 and @tknightly24 can you please share an update where this issue stands? We had discussed that this function would be on the order of 8 lines long (assuming that you used multiple calls to `stringr` functions to pull out different parts and pieces that would be needed. 

If the implementation can't be added to the https://github.com/STAT325-S24/HolyokeUrbanRenewal/blob/main/R/commands.R file, can you please add a description of the documentation and API here?

**Comment by nicholasjhorton** [comment]
Is this helpful in terms of the regular expressions that you were thinking about?

``` r
test <- c("block_01_parcel_06A_info_01.png", "block_02_parcel_06_info_01.png")
stringr::str_replace(stringr::str_extract(test, "block_[0-9]{2}"), "block_", "") # block number
#> [1] "01" "02"
stringr::str_replace(stringr::str_extract(test, "parcel_[0-9A]{2,3}"), "parcel_", "") # parcel
#> [1] "06A" "06"
```

<sup>Created on 2024-02-28 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by arogers24** [comment]
@nicholasjhorton I just finished writing the function. It's [pushed to the repo](https://github.com/STAT325-S24/HolyokeUrbanRenewal/commit/036b1e36e1444e50f30e62e05c91e1e97dde3020). It's a bit messy with regular expressions and substrings, but it should work!

**Comment by nicholasjhorton** [comment]
This is looking good.

One question: shouldn't this function generate the list of filenames from the package?

```
image_path <- paste0(
  system.file(package = "HolyokeUrbanRenewal"),
  "/extdata"
)
files <- list.files(image_path) 
```

Perhaps I'm missing something.

**Comment by arogers24** [comment]
I was working under the assumption that the vector of file names would be a parameter. Is this where we pull the file names?

**Comment by nicholasjhorton** [comment]
Imagine that you are a user who wants to generate a tibble with all of the building info: I think that the function should facilitate this process.

Two other questions:

1. are you adding the `filename` info?
2. what's going on with the `locations` object? It is created but doesn't seem to be used in the function.

**Comment by arogers24** [comment]
Sure thing. I can work on that now. And you're right. I need to add filename, and `locations` can be omitted

**Comment by arogers24** [comment]
[Committed!](https://github.com/STAT325-S24/HolyokeUrbanRenewal/commit/500d77ce5aabe0b8ab367c3192e1bc8cd6607ca7) Let me know if there's anything else

**Comment by nicholasjhorton** [comment]
Alas, things aren't working on my end. How are you testing it? 

Can you please that you've been able to successfully run `roxygen2::roxygenize()` for the package? When I do so I get the following error message:

```
! Failed to load R/commands.R
Caused by error in `get_buildings() <- (function() {
    result <- tibble(filename = character(), block = numeric(), parcel = character(), type = character(),
    file_number = numeric())
    file_names <- list.files("./inst/extdata/", "block")
    result$filename <- file_names
    result$block <- substr(file_names, 7, 8)
    parcel_to_end_locations <- str_locate(file_names, "parcel_[0-9A]{2,3}_")
    result$parcel <- substr(file_names, 17, parcel_to_end_locations[, 2] - 1)
    type_to_end_locations <- str_locate(file_names, "_[a-z]*_[0-9]{2}.png$")
    type_to_end <- substr(file_names, type_to_end_locations[, 1], type_to_end_locations[, 2])
    type_locations <- str_locate(type_to_end, "_[a-z]*_")
    result$type <- substr(type_to_end, type_locations[, 1] + 1, type_locations[, 2] - 1)
    result$file_number <- substr(file_names, str_length(file_names) - 5, str_length(file_names) - 4)
    return(result)
  })`:
! invalid (NULL) left side of assignment
Run `rlang::last_trace()` to see where the error occurred
```

**Comment by nicholasjhorton** [comment]
See https://github.com/STAT325-S24/HolyokeUrbanRenewal/commit/c98cf7cba3b3cde1aed47afbe4f26ce1a6532902 for a proposed fix: can you please review and let me know if you see any issues?

**Comment by nicholasjhorton** [comment]
The call to `system.file()` was left off: I'm puzzled as to how this was working on your end. 

See https://github.com/STAT325-S24/HolyokeUrbanRenewal/commit/dd2c81fb9370e598c4318a30c92a3e1c6f1af6b0 for what I think will be needed.

**Comment by arogers24** [comment]
Yes, I just noticed that issue too. There is another issue, but I'm afraid there's been issue with Git, and I'm afraid to push anything. 

We need to fix the tibble length. We can do so with the following changes:

```
file_names <- list.files("./inst/extdata/", "block")
  
  result <- tibble(
    filename = character(length(file_names)),
    block = numeric(length(file_names)),
    parcel = character(length(file_names)),
    type = character(length(file_names)),
    file_number = numeric(length(file_names))
  )

**Comment by nicholasjhorton** [comment]
Here's a fix which generates the following output:

```
> get_buildings() |> filter(parcel == "06")
# A tibble: 42 × 5
   filename                        block parcel type  file_number
   <chr>                           <chr> <chr>  <chr> <chr>      
 1 block_03_parcel_06_info_01.png  03    06     info  01         
 2 block_03_parcel_06_map_01.png   03    06     map   01         
 3 block_03_parcel_06_map_02.png   03    06     map   02         
 4 block_03_parcel_06_photo_01.png 03    06     photo 01         
 5 block_04_parcel_06_info_01.png  04    06     info  01         
 6 block_04_parcel_06_map_01.png   04    06     map   01         
 7 block_04_parcel_06_photo_01.png 04    06     photo 01         
 8 block_05_parcel_06_info_01.png  05    06     info  01         
 9 block_05_parcel_06_map_01.png   05    06     map   01         
10 block_05_parcel_06_photo_01.png 05    06     photo 01         
# ℹ 32 more rows
# ℹ Use `print(n = ...)` to see more rows
> get_buildings() |> filter(parcel == "06A")
# A tibble: 3 × 5
  filename                         block parcel type  file_number
  <chr>                            <chr> <chr>  <chr> <chr>      
1 block_06_parcel_06A_info_01.png  06    06A    info  01         
2 block_06_parcel_06A_map_01.png   06    06A    map   01         
3 block_06_parcel_06A_photo_01.png 06    06A    photo 01         
> 
```

**Comment by nicholasjhorton** [comment]
Here's a link to my commit: https://github.com/STAT325-S24/HolyokeUrbanRenewal/commit/7e19d49b46d84c4428d6e602df41d90b64b653fd

and to the current contents of https://github.com/STAT325-S24/HolyokeUrbanRenewal/blob/main/R/commands.R

I believe that this is now working as expected so will close the issue.

Can everyone please pull, check, and install the package to ensure that what you have is up to date and working?
